### PR TITLE
Unit 2 sakaa killen

### PR DESCRIPTION
**Pull request by Sakaa4Short**
# Submitting Unit 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

I would really appreciate feed back on the coding challenges. I struggled to figure out a couple of aspects but I got one of them down for the most part.

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -34,10 +34,31 @@ const object2 = { number: 0 }
...
-// Pseudo code:
-
+describe("divisibleOrNah", () => {
+    it(" decides if the number inside it is evenly divisible by three or not", () => {
+        expect(divisibleOrNah(15)).toEqual("15 is divisible by three")
+        expect(divisibleOrNah(0)).toEqual("0 is divisible by three")
+        expect(divisibleOrNah(-7)).toEqual("-7 is not divisible by three")
+    })
+
+})
```
Comment (sjproctor) [inline code-challenges.test.js:44]: The purpose of the test variables is to give you specific input. In this case, you didn't use the test variable in your expect statement which means your code would have falsely passed the test.

File: code-challenges.test.js [hunk 2]
```diff
@@ -34,10 +34,31 @@ const object2 = { number: 0 }
...
+// // Create a test that can return the output that whether or not the objects are divisible by three
+
+const divisibleOrNah = (number) => {
+    if (number % 3 === 0 )
+    return `${number} is divisible by three`
+    else {
+        return `${number} is not divisible by three`
+    }
+  }
+
```
Comment (sjproctor) [inline code-challenges.test.js:60]: This code would work with numbers but not the given data type of object.

File: code-challenges.test.js [hunk 3]
```diff
@@ -47,6 +68,38 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
+describe("capitalLetters") = () => {
+    it("return strings with the first letter capitalized", () => {
+        expect(capitalLetters(randomNouns1)).toEqual("Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew")
+        expect(capitalLetters(randomNouns2)).toEqual("Temperature", "Database", "Chopsticks", "Mango")
+    })
+
+}
```
Comment (sjproctor) [inline code-challenges.test.js:77]: This test looks better, but your test should contain all the info it needs inside the scope {} of the describe. Which means your test variables should live inside the test.

File: code-challenges.test.js [hunk 4]
```diff
@@ -47,6 +68,38 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
+
+const capitalLetters = (array) => {
+    return array.map((value) => {
+      if (typeof value === 'string' || value instanceof String) {
+        return value.charAt(0).toUpperCase() + value.slice(1)
+      } else {
+        return value
+      }
+    })
+}
```
Comment (sjproctor) [inline code-challenges.test.js:97]: I wonder if the additional edge case checks on line 91 are interfering with your code. Line 92 would take care of the necessary logic.

File: coding-concepts.js [hunk 5]
```diff
@@ -9,28 +9,29 @@
...
 
 // --------------------2) What will this log?
 
 const greeter = (name) => {
   `Hello, ${name}!`
 }
-// console.log(greeter("LEARN Student"))
+console.log(greeter("LEARN Student"))
+
+// a) Your answer: There is no return so the terminal will give no answer.
```
Comment (sjproctor) [inline coding-concepts.js:24]: Nice!

**Review by sjproctor — APPROVED** [review]
Nice work! Your interview and coding concepts sections are great! Your coding is coming along. I encourage you to follow your instincts rather than reaching for external help that ends up complicating your code unnecessarily. You have an advanced beginner level of skill on this assessment.
